### DISCUSSION TITLE
Does `retry_on_initial_connect` retry forever?

### DISCUSSION THREAD

**Discussion by feelingsonice** [nats-io/nats.rs#1312] [Q&A] [answered]
Don't see any timeouts or retry limit for it, does it retry forever if I set it to `true`?

**Comment by Jarema** [comment, marked as answer]
Yes, it retires forever with exponential backoff.
You can set limit with [max_reconnects](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.max_reconnects)

To customize the reconnect delays use [reconnect_delay_callback](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.reconnect_delay_callback)

**Reply by feelingsonice** [reply]
Does `max_reconnects` also affect reconnecting _after_ the initial connection?

**Reply by Jarema** [reply]
yes. Its maximum number of attemps after loosing connection. Counter will be reset after successful reconnect attempt.

from docs:
```
    /// Specifies the number of consecutive reconnect attempts the client will
    /// make before giving up. This is useful for preventing zombie services
    /// from endlessly reaching the servers, but it can also be a footgun and
    /// surprise for users who do not expect that the client can give up
    /// entirely.
    ///
    /// Pass `None` or `0` for no limit.
    ///
```
